### PR TITLE
Use a later version of requests package to address a CVE

### PR DESCRIPTION
**Pull request by PolakAllen**
Not sure why this was pinned to 2.29.0, but it's causing later versions of the seldon package to be exposed this CVE:

https://nvd.nist.gov/vuln/detail/CVE-2023-32681

I'll be using `2.0.0` in the meantime (which doesn't have the constraint), and let you know if I see any issues with the package, but I can't think of any reason why you'd need this pin.

### CODE REVIEW

**Comment by PolakAllen** [comment]
Just noting that if you have other systems with the constraint, making a wider bound might be good (the one I gave is a bit strict).

For a very permissive example, probably something like this? Hoping we don't need any upper bound.

```toml
"requests>=2.0"
```

File: templates/python/setup.mustache [hunk 1]
```diff
@@ -22,7 +22,7 @@ REQUIRES = [
     "six>=1.10",
     "urllib3>=1.23",
     "Authlib>=1.0.0,<1.1.0",
-    "requests<=2.29.0"
+    "requests>=2.31.0"
```
Comment (jesse-c) [inline templates/python/setup.mustache:25]: Could you apply this patch [1] and then run `make python`, and commit the changes?

```patch
diff --git a/templates/python/setup.mustache b/templates/python/setup.mustache
index 8c06d85..c6b490c 100644
--- a/templates/python/setup.mustache
+++ b/templates/python/setup.mustache
@@ -22,7 +22,7 @@ REQUIRES = [
     "six>=1.10",
     "urllib3>=1.23",
     "Authlib>=1.0.0,<1.1.0",
-    "requests>=2.31.0"
+    "requests >= 2.0.0, <= 3.0.0"
 ]
 
 {{#asyncio}}
```

**Review by jesse-c — CHANGES_REQUESTED** [review]
Hi @PolakAllen! Thanks for the report and the PR. You're right that we could be more permissive, and in fact, we are with `requirements`(https://github.com/SeldonIO/seldon-deploy-sdk/blob/master/templates/python/requirements.mustache#L4) file.

File: templates/python/setup.mustache (hunk 1 above)
Comment (PolakAllen) [reply templates/python/setup.mustache:25]: Used `<3.0` for the upper bound, but otherwise as requested.

Comment (PolakAllen) [reply templates/python/setup.mustache:25]: Sorry, one more try on my end, lost my change somehow. Also I now see the `requirements.txt`, so I'll use that exactly.

Comment (jesse-c) [reply templates/python/setup.mustache]: No worries! Yeah, there's the templating step which isn't immediately obvious!

**Review by jesse-c — APPROVED** [review]
LGTM

We (Seldon) will do an RC release after it's merged. I'll let you know once it's been done.

Thanks for working on this!

**Comment by jesse-c** [comment]
> The base branch requires all commits to be signed.

Are you able to sign the 2 latest commits or squash them into your first signed commit?

**Comment by PolakAllen** [comment]
> > The base branch requires all commits to be signed.
> 
> Are you able to sign the 2 latest commits or squash them into your first signed commit?

I'll squash + look into signing when I get home, thanks

**Comment by PolakAllen** [comment]
Hmm, git doesn't want to accept my commit as verified for some reason. It's still unclear why.

I've added my gpg key directly to my account, and the email should match.

I'll give it some time in case it's a sync issue.

**Comment by jesse-c** [comment]
@PolakAllen: Sounds good. From my experience, it should be quick, but you never know with technology. ;) FWIW, your first commit in this PR was signed and displaying as verified.

**Review by jesse-c — APPROVED** [review]

**Comment by jesse-c** [comment]
@PolakAllen: I could probably push a branch to this repository, cherry-picking your commit, and amend the commit to sign it myself.

**Comment by jesse-c** [comment]
@PolakAllen: Thank you for the original work on this! To speed up delivery of the fix for the CVE, we've gone with https://github.com/SeldonIO/seldon-deploy-sdk/pull/69. We'll let you know once there's a release for you to use!

**Comment by jesse-c** [comment]
We've released https://pypi.org/project/seldon-deploy-sdk/2.2.0rc9/.

**Pull request closed without merging**